guidance/library/_json.py

```
@@ -94,53 +96,42 @@ def json(
-    # you call `json` instead of waiting for generation to fail.
-    VALIDATE = True
-    if VALIDATE:
-        schema_string = json_dumps(schema)
-        try:
-            compiler = JsonCompiler(
-                separators=separators,
-                whitespace_flexible=whitespace_flexible,
-                coerce_one_of=False,
-            )
-            compiler.compile(schema_string)
-        except ValueError as e:
-            if (
-                e.args[0]
-                == "oneOf constraints are not supported. Enable 'coerce_one_of' option to approximate oneOf with anyOf"
-            ):
-                warnings.warn(
-                    "oneOf not fully supported, falling back to anyOf. This may cause validation errors in some cases."
-                )
-                compiler = JsonCompiler(
-                    separators=separators,
-                    whitespace_flexible=whitespace_flexible,
-                    coerce_one_of=True,
-                )
-                compiler.compile(schema_string)
-                coerce_one_of = True
-            else:
-                raise
+    if whitespace_pattern is not None:
+        whitespace_flexible = True
 
     if separators is None:
-        if whitespace_flexible:
+        if whitespace_flexible or whitespace_pattern:
             separators = (",", ":")
         else:
             separators = (", ", ": ")
+    if len(separators) != 2:
+        raise ValueError("separators must be a tuple of (item_separator, key_separator)")
     item_separator, key_separator = separators
 
     if schema.get("x-guidance") is None:
         schema["x-guidance"] = {
             "item_separator": item_separator,
             "key_separator": key_separator,
             "whitespace_flexible": whitespace_flexible,
-            "coerce_one_of": coerce_one_of,
+            "whitespace_pattern": whitespace_pattern,
+            "coerce_one_of": True,
+            "lenient": lenient,
```

@mmoskal I'm open to changing guidance's behavior, dropping `coerce_one_of=True` and using your `lenient=False` default. Will probably break some tests, so you're fine to leave this how it is -- more of a mental not to myself to address this later.

---

This setting (when lenient=False) simulates previous behavior: oneOf is allowed, but produces a warning (previously you would try without coercion, and if it failed print warning)

---

Yep, makes sense!

---

Oh. I see. You just said to not do the warning stuff by default but fail loudly. Up to you I guess!